pkg/logs/sender/connection_manager.go

```diff
@@ -53,37 +52,55 @@ func (cm *ConnectionManager) NewConnection() net.Conn {
 	cm.mutex.Lock()
 	defer cm.mutex.Unlock()
 
+	var retries int
 	for {
 		if cm.firstConn {
-			log.Info("Connecting to the backend: ", cm.connectionString)
+			if cm.proxyAddress != "" {
```

Comment:
could this `if cm.firstConn` be moved out of the for loop? It should happen once only